tests/src/test/java/alluxio/testutils/underfs/sleeping/SleepingUnderFileSystemFactory.java

```diff
@@ -41,14 +42,17 @@ public SleepingUnderFileSystemFactory(SleepingUnderFileSystemOptions options) {
    * @param ufs the UFS instance to return on creation
    */
   public SleepingUnderFileSystemFactory(SleepingUnderFileSystem ufs) {
-    mUfs = ufs;
+    mUfs = new ManagedBlockingUfsForwarder(ufs);
   }
 
   @Override
   public UnderFileSystem create(String path, UnderFileSystemConfiguration conf) {
     if (mUfs == null) {
       Preconditions.checkArgument(path != null, "path may not be null");
-      return new SleepingUnderFileSystem(new AlluxioURI(path), mOptions, conf);
+      // Wrapping with a special forwarder to let master thread-pool know that threads will sleep
+      // on this UFS.
+      return new ManagedBlockingUfsForwarder(
+          new SleepingUnderFileSystem(new AlluxioURI(path), mOptions, conf));
```

Comment:
Instead of directly wrapping, should we modify the conf to set managedblocking to true?

Reply:
You mean overriding the conf within `SleepingUnderFileSystemFactory`? Or in all tests that use `SleepingUnderFileSystem` ?

Reply:
overriding the conf within `SleepingUnderFileSystemFactory` if it is not set

Reply:
I feel uneasy about introducing a special config override.  Could you please explain what do you mean by ufs not being automatically wrapped?

Reply:
Currently the managed blocking flag is turned on if `isObjectStore` is true or if the managed blocking configuration value is set. I would prefer to use one mechanism to determine if we are using managed blocking.

What I mean is - if somewhere in our code we get a new S3AUnderFileSystem, it will not be wrapped in managed blocking (only if it is created through the UfsManager).

Reply:
@gpang right, it would be set in the ufsconf, ideally we want to check if it has been set already (not sure if we can distinguish between set and default value though)

Reply:
Understood. PTAL at the change.